client/relay.go

```diff
@@ -12,35 +12,38 @@ import (
 
 var _ sdk.RelayClient = (*relayClient)(nil)
 
-
+// relayClient is a RelayClient implementation that uses the HTTP client
+// of the Go standard library.
 type relayClient struct {
 	httpClient *http.Client
 }
 
+// NewRelayClient creates a new relay client with the default HTTP client, which
+// is used to send the relay requests.
 func NewRelayClient() (sdk.RelayClient, error) {
 	return &relayClient{
 		httpClient: http.DefaultClient,
 	}, nil
 }
 
+// SendRequest sends the relay request to the supplier at the given URL.
+// It accepts the relay request bytes to avoid relying on protocol specific
+// request objects such as http.Request.
 func (r *relayClient) SendRequest(
 	ctx context.Context,
 	supplierUrlStr string,
-	relayRequestBz []byte,
-	method string,
-	requestHeaders map[string][]string,
+	requestBz []byte,
```

Comment:
This is fine for now, but we should add a TODO to add support for the different protocols IN THE SDK (grpc, websockets, etc...)

Reply:
`RelayClient` and its method `SendRequest` is about the communication between the `SDK` (`AppGateServer`) and the `RelayMiner`, which does not necessarily need to happen using the same protocol or `RPCType` as the upstream request. The only exception would be for `websockets` and other realtime/async `RPCType`s

In fact the upstream request is serialized to bytes, embedded into a `RelayRequest` then transmitted to the `RelayMiner`. Same for the response.

The `SDK` has no knowledge about what kind of data is being transmitted nor which protocol or `RPCType` is being used.

I still believe that the `SDK` should offer helpers to build such gateways instead of only facilitating the communication between the `Gateway` that uses it and the `RelayMiner`. Which is what it is currently about.

To summerize
* The `AppGateServer` needs to have the right "server", to handle different `RPCType`s
* The `RelayMiner` needs to have the right "client", to forward requests to the `BackendService`
* We are in control of how `AppGateServer`'s client and `RelayMiner`'s server communicate
  * I don't think we should offer diversity here. But would love to go realtime gRPC or pure TCP
  * Will add documentation about this (in the `poktroll` repo)
* Will add the relevant `TODO`s here

Reply:
I like the levels of abstraction, but I think it's okay for the SDK to have more. Like I said in the other PR, if both nodies and grove will need to reimplement it, move it to the SDK. We can still keep the same level of abstraction at the interface layer.